examples/lorawan/riot.yaml

```diff
@@ -0,0 +1,36 @@
+version: 0.1.0
+name: lorawan-example
+authors:
+- name: Alexandre Abadie
+  email: alexandre.abadie@inria.fr
+license: LGPL-2.1
+copyright: [2018 Inria]
+configuration:
+- group:
```

Comment:
The number of nesting levels and the verbosity can be reduced by using dicts of dicts instead of lists of dicts:
```yaml
version: 0.1.0
name: lorawan-example
authors:
- name: Alexandre Abadie
  email: alexandre.abadie@inria.fr
license: LGPL-2.1
copyright: [2018 Inria]
configuration:
  lorawan: # group disappears
    appkey: # name disappears and so does macro
      description: LoRaWAN Application Key
      value: 0xAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
      type: byte
      properties:
        size: 16
    appeui:
      description: LoRaWAN Application EUI
      value: 0xBBBBBBBBBBBBBBBB
      type: byte
      properties:
        size: 8
    deveui:
      description: LoRaWAN Device EUI
      value: 0xCCCCCCCCCCCCCCCC
      type: byte
      properties:
        size: 8
  period:
    description: Seconds to wait between messages
    type: int
    value: 20
```

Reply:
open question with this approach: how do you know some name is a an option and not a namespace for other options without complicated guessing. Maybe option containers can be prefixed with a character (@ ??) or start with a capital letter.